elastic/eui
# Fix/improve Typescript references in all Cypress test files

## Summary

While writing some recent EuiComboBox Cypress tests, I got super annoyed at how badly Cypress/Typescript was working in my local VSCode IDE - API autocompletes were constantly wrong, I was getting multiple `Cannot find name 'cy'` errors in my problems pane, etc.

<img width="247" alt="" src="https://user-images.githubusercontent.com/549407/230498565-8d1f79f8-0dfe-4cc5-88b3-3a374c4f8af4.png">

After some extremely dumb luck, I figured out that the issues only appear if [cypress/support/index.d.ts](https://github.com/elastic/eui/blob/bb8fefa12eceb1970585943bf2a0266473c0b5a6/cypress/support/index.d.ts) is not **open in a tab in VSCode** on load 🤦

`/// <reference types="cypress" />` "forces" the errors to at least stop appearing, so I've created an eslint rule for that in all `*.spec.tsx` and `*.a11y.tsx` files with an autofixer (2nd commit & 3rd). This should at least get us to not having a million problems as well as getting us intellisense/autocomplete.

Unfortunately, the references don't seem to fix type checking on our source components no matter what I try to do with `tsconfig.json` - for some bizarre reason, `cypress/support/index.ts` _has_ to be open for that to work 100% correctly. I've given up on that for now so that's still semi-broken 🤷 

I also cleaned up and DRY'd out our setup (4th & 5th commit) while here.

## QA

- Check out branch
- Ensure `cypress/support/index.d.ts` is not open in a tab (close it if it is)
- Restart your IDE
- Open up any random `.spec.tsx` file
- [x] Confirm no errors appear in the problems pane and `cy.mount` autofills if you start typing `cy.m`

### General checklist

N/A - dev only
